client/ayon_hiero/plugins/publish/collect_plates.py

```diff
@@ -39,12 +40,45 @@ def process(self, instance):
         if review_switch is True:
             if reviewable_source == "clip_media":
                 instance.data["families"].append("review")
+                instance.data.pop("reviewTrack", None)
+
+                # Retrieve source clip media from otio clip.
+                # image sequence
+                if hasattr(otio_clip.media_reference, "target_url_for_image_number"):
+                    file_url = otio_clip.media_reference.target_url_for_image_number(0)
+                    sequence_length = (
+                        otio_clip.media_reference.end_frame()
+                        - otio_clip.media_reference.start_frame
+                    )
+                    file_names = [
+                        os.path.basename(
+                            otio_clip.media_reference.target_url_for_image_number(frame)
+                        )
+                        for frame in range(0, sequence_length)
+                    ]
+
+                # movie
+                else:
+                    file_url = otio_clip.media_reference.target_url
+                    file_names = os.path.basename(file_url)
+
+                _, ext = os.path.splitext(file_url)
+
+                repre = {
+                    "name": ext[1:],
+                    "ext": ext[1:],
+                    "files": file_names,
+                    "stagingDir": os.path.dirname(file_url),
+                    "tags": ["review", "delete"]
+                }
+                instance.data["representations"].append(repre)
```

Comment:
We are already having a dedicated plugin for collecting representations from auto clip media reference with proper frame ranges. I wonder whether we should not to distribute any way the `review` tag flagging. Here is the code https://vscode.dev/github/ynput/ayon-core/blob/bugfix/fix_extract_otio_review/client/ayon_core/plugins/publish/collect_otio_subset_resources.py#L207

Reply:
Good call, please see https://github.com/ynput/ayon-core/pull/1036.